src/sentry/workflow_engine/migrations/0012_migrate_metric_alerts.py

```diff
@@ -0,0 +1,206 @@
+    AlertRuleProjects = apps.get_model("sentry", "AlertRuleProjects")
+    AlertRuleTrigger = apps.get_model("sentry", "AlertRuleTrigger")
+    AlertRuleTriggerAction = apps.get_model("sentry", "AlertRuleTriggerAction")
+    # AlertRuleActivity = apps.get_model("sentry", "AlertRuleActivity")
+    QuerySubscription = apps.get_model("sentry", "QuerySubscription")
+    Incident = apps.get_model("sentry", "Incident")
+
+    DataSource = apps.get_model("sentry", "DataSource")
+    DataConditionGroup = apps.get_model("sentry", "DataConditionGroup")
+    Workflow = apps.get_model("sentry", "Workflow")
+    Detector = apps.get_model("sentry", "Detector")
+    DetectorState = apps.get_model("sentry", "DetectorState")
+    DataCondition = apps.get_model("sentry", "DataCondition")
+    Action = apps.get_model("sentry", "Action")
+    DataConditionGroupAction = apps.get_model("sentry", "DataConditionGroupAction")
+    DataSourceDetector = apps.get_model("sentry", "DataSourceDetector")
+    DetectorWorkflow = apps.get_model("sentry", "DetectorWorkflow")
+    WorkflowDataConditionGroup = apps.get_model("sentry", "WorkflowDataConditionGroup")
+
+    for rule in RangeQuerySetWrapperWithProgressBarApprox(AlertRule.objects.all()):
+        if rule.status in [AlertRuleStatus.DISABLED, AlertRuleStatus.SNAPSHOT]:
+            continue
+        # what about the date_added and date_updated fields? can I write to those to preserve history?
+        # description column will be added to the Detector model, but at the time of writing it isn't there yet. will need to update this
+        # same for threshold_period, will likely be added to the Detector model but isn't there yet
+        # AlertRuleActivity creation rows need to be backfilled into the Detector model, needs new created_by and created_at columns
+
+        query_subscription = QuerySubscription.objects.get(snuba_query=rule.snuba_query_id)
+        data_source = DataSource.update_or_create(
+            organization_id=rule.organization_id,
+            query_id=query_subscription.id,
+            type=DataSourceType.SNUBA_QUERY_SUBSCRIPTION,
+        )
+        data_condition_group = DataConditionGroup.update_or_create(
+            logic_type=DataConditionGroupType.ANY,
+            organization_id=rule.organization_id,
+        )
+        workflow = Workflow.objects.update_or_create(
+            name=rule.name,
+            organization_id=rule.organization_id,
+            when_condition_group=data_condition_group.id,
+        )
+        detector = Detector.update_or_create(
+            organization_id=rule.organization_id,
+            name=rule.name,
+            data_sources=data_source,
+            workflow_condition_group=data_condition_group.id,
+            type=MetricAlertFire.slug,
+            owner_user_id=rule.user_id,
+            owner_team=rule.team,
+        )
+
+        # state is based on incident status
+        project = AlertRuleProjects.objects.get(alert_rule_id=rule.id)
```

Comment:
are there any performance concerns with us fetching all the alert rules, then fetching these individually?

Reply:
It'll be slower, but we have < 200k metric alerts, so I don't think it's a huge deal overall